docs/en/advanced/neural_networks.md

```diff
@@ -0,0 +1,107 @@
+# Neural Networks
+
+<Badge type="tip" text="main (planned for: PX4 v1.17)" /> <Badge type="warning" text="Experimental" />
+
+::: warning
+This is an experimental module.
+Use at your own risk.
+:::
+
+The Multicopter Neural Network (NN) module  ([mc_nn_control](../modules/modules_controller.md#mc_nn_control)) is an example module that allows you to experiment with using a pre-trained neural network on PX4.
+It might be used, for example, to experiment with controllers for non-traditional drone morphologies, computer vision tasks, and so on.
+
+The module integrates a pre-trained neural network based on the [TensorFlow Lite Micro (TFLM)](../advanced/tflm.md) module.
+The module is trained for the [X500 V2](../frames_multicopter/holybro_x500v2_pixhawk6c.md) multicopter frame.
+While the controller is fairly robust, and might work on other platforms, we recommend [Training your own Network](#training-your-own-network) if you use a different vehicle.
+Note that after training the network you will need to update and rebuild PX4.
+
+TLFM is a mature inference library intended for use on embedded devices.
+It has support for several architectures, so there is a high likelihood that you can build it for the board you want to use.
+If not, there are other possible NN frameworks, such as [Eigen](https://eigen.tuxfamily.org/index.php?title=Main_Page) and [Executorch](https://pytorch.org/executorch-overview).
+
+The document explains how you can include the module in your PX4 build, and provides a broad overview of how it works.
+The other documents in the section provide more information about the integration, allowing you to replace the NN with a version trained on different data, or even to replace the TLFM library altogether.
+
+## Neural Network PX4 Firmware
+
+The module has been tested on the a number of configurations, which can be build locally using the commands:
+
+```sh
+make px4_sitl_neural
+```
+
+```sh
+make px4_fmu-v6c_neural
+```
+
+```sh
+make mro_pixracerpro_neural
+```
+
+You can add the module to other board configurations by modifying their `default.px4board file` configuration to include these lines:
+
+```sh
+CONFIG_LIB_TFLM=y
+CONFIG_MODULES_MC_NN_CONTROL=y
+```
+
+## Example Module Overview
```

Comment:
It wasn't clear if "this is what you can do" or "this is what I did/what the module does".

I think it is "This is what the module does now, to help you understand how it all works". Might need a little more work to make this consistent. Depending on what you say.